src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,40 @@
+package mate.academy.dao;
+
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+
+    @Override
+    public Movie add(Movie movie) {
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            Transaction transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (HibernateException e) {
+            throw new DataProcessingException("cant add new movie " + movie, e);
+        } finally {
+            session.close();
+        }
+        return movie;
```

Comment:
you return movie without initialized field id here

Reply:
> you return movie without initialized field id here

I think hibernate assigns id by itself

Reply:
I've already checked it, it is returned movie with an id field